component/client/ClientChat.jsx

```diff
@@ -1,14 +1,32 @@
 import React, {Component} from "react";
 import {Panel} from "react-bootstrap";
 import ClientMessenger from "./ClientMessenger.jsx";
+import RoomService from "./RoomService";
 
 class ClientChat extends Component {
+
+  room: RoomService;
+
   constructor(props) {
     super(props);
+    this.state = {
+      history: []
+    };
+    // set up Room Service
+    this.room = new RoomService(props.url, (message) => {
```

Comment:
componentWillReceiveProps(nextProps) is a good idea.
I don't think redux is necessary here. Firebase is fast enough for you to recreate this service every time. It would only load messages in this room(url), i.e. O(n), the same big-O as updating UI. There is no need to store messages of a different url.
Firebase database is a singleton, it will not be recreated every time, creating a reference is O(1).